### ISSUE TITLE
env['warden'].session(:sudo)[:foo] = "bar" doesn't set "bar"

### ISSUE THREAD

**Issue by zjeraar** [hassox/warden#134]
If `raw_session["warden.user.sudo.session"]` is not set, this `env['warden'].session(:sudo)[:foo] = "bar"` doesn't set `raw_session["warden.user.sudo.session"][:foo]` to "bar", but only creates the hash. You'd have to do:

```
env['warden'].session(:sudo)
env['warden'].session(:sudo)[:foo] = "bar"
```

**Comment by halo** [comment]
The problem you describe does not seem to happen to me. Also, [the code indicates](https://github.com/hassox/warden/blob/906edf86c6c31be917a921097031b89361d022e8/lib/warden/proxy.rb#L234) that it should work without calling twice. However, [I cannot find any](https://github.com/hassox/warden/blob/906edf86c6c31be917a921097031b89361d022e8/spec/warden/proxy_spec.rb) specs for the `Proxy#session` method, where are they?

Because this is what I experience: when I set something in Rails with 

```ruby
request.env['warden'].session(:admin)[:thekey) = 'thevalue'
```
this is what I get: 
```ruby
{ 'thekey' => 'thevalue' }
# I expected though:
# { thekey: 'thevalue' }
```
in other words, the keys are stringified. Now this can be something I messed up in my setup, but I'm still curious whether this happens to others as well. Either way, if someone can point me to the test, I'd be glad.
